library/modules/EventManager.cpp

```diff
@@ -250,6 +283,19 @@ void DFHack::EventManager::onStateChange(color_ostream& out, state_change_event
         if (!df::global::world)
             return;
 
+        std::string save_dir = "data/save/" + df::global::world->cur_savegame.save_dir;
+        //std::string savegame = save_dir + "/world.sav";
+        std::string persistent_file = save_dir + "/dfhack_persistent_data.dat";
+        DFHack::Persistent::clear(); //just in case
+        DFHack::Persistent::readFromFile(persistent_file);
+        /*savetime = DFHack::Filesystem::mtime(savegame);
+        std::string persistent_file = save_dir + "/dfhack_persistent_data.dat";
+        if ( DFHack::Filesystem::exists(persistent_file) ) {
+            boost::property_tree::json_parser::read_json(persistent_file, persistent_tree);
+        } else {
+            persistent_tree = pt::ptree();
+        }*/
+
```

Comment:
You seem to be adding a few chunks of commented-out code near here.  Are they ideals to work toward in the future, or should they be deleted?